src/nvim/highlight.c

```diff
@@ -177,8 +177,22 @@ void update_window_hl(win_T *wp, bool invalid)
   }
 }
 
+/// Gets HL_UNDERLINE highlight.
+int hl_get_underline(void)
+{
+  int hl_attrs = HL_UNDERLINE;
+  return hl_get_term_attr(&(HlAttrs) {
```

Comment:
using `kHlTerminal` here. Not sure if it matters.

Reply:
should be `kHlUI` if the trick is used, but I think it can be avoided entirely (see other comment).